File: Sesion 6/Marco teorico/18CÓMO SE SOBRECARGAN LOS OPERADORES BINARIOS/18.java
class Punto {
    private double x;
    private double y;

    Punto(double x, double y) {
        this.x = x;
        this.y = y;
    }

    Punto sumar(Punto otro) {
        return new Punto(this.x + otro.x, this.y + otro.y);
    }

    void mostrar() {
        System.out.println("(" + x + ", " + y + ")");
    }
}

public class Main {
    public static void main(String[] args) {
        Punto punto1 = new Punto(1.0, 2.0);
        Punto punto2 = new Punto(3.0, 4.0);

        Punto suma = punto1.sumar(punto2);

        System.out.print("La suma de los puntos es: ");
        suma.mostrar();
    }
}

File: Sesion 6/Marco teorico/18CÓMO SE SOBRECARGAN LOS OPERADORES BINARIOS/18.py
class Punto:
    def __init__(self, x=0.0, y=0.0):
        self.x = x
        self.y = y

    def __add__(self, otro):
        return Punto(self.x + otro.x, self.y + otro.y)

    def mostrar(self):
        print("(", self.x, ",", self.y, ")")


punto1 = Punto(1.0, 2.0)
punto2 = Punto(3.0, 4.0)

suma = punto1 + punto2

print("La suma de los puntos es: ", end="")
suma.mostrar()